packages/astro/src/cli/check/index.ts

```diff
@@ -1,101 +1,360 @@
 /* eslint-disable no-console */
-import { AstroCheck, DiagnosticSeverity } from '@astrojs/language-server';
+import { AstroCheck, DiagnosticSeverity, GetDiagnosticsResult } from '@astrojs/language-server';
 import type { AstroSettings } from '../../@types/astro';
 import type { LogOptions } from '../../core/logger/core.js';
-
 import glob from 'fast-glob';
 import * as fs from 'fs';
 import { bold, dim, red, yellow } from 'kleur/colors';
 import { createRequire } from 'module';
 import ora from 'ora';
 import { fileURLToPath, pathToFileURL } from 'url';
 import { printDiagnostic } from './print.js';
+import type { Arguments as Flags } from 'yargs-parser';
+import { debug, info } from '../../core/logger/core.js';
+import type { SyncOptions, ProcessExit } from '../../core/sync';
```

Comment:
This is not a problem because it's a type only import, but we typically only use `.js` imports

Reply:
I don't think we've really standardized that yet. For type imports I often omit it. If we're all doing it differently we should pick one, I'd be happy with either way. Maybe always using `.js` safer.

Reply:
Considering we're forced to use `.js` imports for normal imports, imo we should do the same for all imports

Reply:
`import type` is elided at compilation time, so I thought it didn't matter, and I let the IDE do its thing without thinking about it. I can add `.js` for consistency!

Reply:
It turns out that adding `.js` emits the TS error... I have no idea why.